### PR TITLE
Update dm_mripicture.py

### PR DESCRIPTION
**Pull request by kyjimmy**
add logging module and skip subjects with existing files

### CODE REVIEW

File: bin/dm_mripicture.py [hunk 1]
```diff
@@ -14,11 +14,16 @@
     -s --subject        Subjects
     -o --output=FOLDER  Output directory
     -t --tag=TAG        Scan tag [default: T1]
+    -r --replace        Replace existing files [default: False]
```
Comment (josephmje) [inline bin/dm_mripicture.py:17]: Suggesting changing this to `--force` to align with other typical command line interfaces.

```suggestion
    -f --force        Force overwrite of output files [default: False]
```

File: bin/dm_mripicture.py [hunk 2]
```diff
@@ -27,6 +32,11 @@
 import datman.scan
 
 
+logging.basicConfig(level=logging.WARN,
+                    format="[%(name)s] %(levelname)s: %(message)s")
```
Comment (josephmje) [inline bin/dm_mripicture.py:36]:
```suggestion
```

File: bin/dm_mripicture.py [hunk 3]
```diff
@@ -40,20 +50,35 @@ def main():
     outdir = arguments['--output']
     subs = arguments['<subject>']
     tag = arguments['--tag']
+    overwrite = arguments['--replace']
```
Comment (josephmje) [inline bin/dm_mripicture.py:53]:
```suggestion
    force = arguments['--force']
```

File: bin/dm_mripicture.py [hunk 4]
```diff
@@ -40,20 +50,35 @@ def main():
...
+    debug = arguments['--debug']
+    verbose = arguments['--verbose']
     config = datman.config.config(study=study)
 
+    if verbose:
+        logger.setLevel(logging.INFO)
+    if debug:
+        logger.setLevel(logging.DEBUG)
+    if quiet:
+        logger.setLevel(logging.ERROR)
```
Comment (josephmje) [inline bin/dm_mripicture.py:64]:
```suggestion
    # setup logging
    ch = logging.StreamHandler(sys.stdout)
    ch.setLevel(logging.WARN)
    logger.setLevel(logging.WARN)
    if quiet:
        logger.setLevel(logging.ERROR)
        ch.setLevel(logging.ERROR)
    if verbose:
        logger.setLevel(logging.INFO)
        ch.setLevel(logging.INFO)
    if debug:
        logger.setLevel(logging.DEBUG)
        ch.setLevel(logging.DEBUG)

    formatter = logging.Formatter(f"%(asctime)s - %(name)s - {study} - "
                                  "%(levelname)s - %(message)s".)
    ch.setFormatter(formatter)

    logger.addHandler(ch)
```

File: bin/dm_mripicture.py [hunk 5]
```diff
@@ -40,20 +50,35 @@ def main():
...
-        print('Creating brain pictures for all subjects from',
-              study, 'project using', tag, 'scans.')
+        logger.info('Creating pictures for all {} subjects from {} project '
+                    'using {} scans.'.format(len(subs), study, tag))
 
     if not outdir:
         outdir = os.path.join(config.get_path('data'), 'tshirt')
 
     os.makedirs(outdir, exist_ok=True)
+    logger.debug('Output location set to: {}'.format(outdir))
```
Comment (josephmje) [inline bin/dm_mripicture.py:78]:
```suggestion
    logger.debug(f'Output location set to: {outdir}')
```

File: bin/dm_mripicture.py [hunk 6]
```diff
@@ -65,14 +90,20 @@ def main():
...
-        t1_pic = plotting.plot_anat(imgpath, cut_coords=(-20, -10, 2),
-                                    display_mode='x', annotate=False,
-                                    draw_cross=False, vmin=100,
-                                    vmax=1100, threshold='auto')
-        t1_pic.savefig(outpath, dpi=1000)
-
-    print('Saved all output to', outdir)
+        if os.path.isfile(outpath) and not overwrite:
+            logger.debug('Skipping subject {} as files already exist.'
+                         .format(subject))
```
Comment (josephmje) [inline bin/dm_mripicture.py:95]:
```suggestion
            logger.debug(f'Skipping subject {subject} as files already exist.')
```

File: bin/dm_mripicture.py [hunk 7]
```diff
@@ -65,14 +90,20 @@ def main():
...
+        else:
+            # Output Image
+            t1_pic = plotting.plot_anat(imgpath, cut_coords=(-20, -10, 2),
+                                        display_mode='x', annotate=False,
+                                        draw_cross=False, vmin=100,
+                                        vmax=1100, threshold='auto')
+            t1_pic.savefig(outpath, dpi=1000)
+            logger.debug('Created new brain pictures for subject {} from file '
+                         '{} and saved as {}'.format(subject, imgpath, outpath))
```
Comment (josephmje) [inline bin/dm_mripicture.py:104]:
```suggestion
            logger.debug(f'Created new brain pictures for subject {subject} from file '
                         f'{imgpath} and saved as {outpath}')
```

File: bin/dm_mripicture.py [hunk 8]
```diff
@@ -65,14 +90,20 @@ def main():
...
+
+    logger.info('Saved all output to: {}'.format(outdir))
```
Comment (josephmje) [inline bin/dm_mripicture.py:106]:
```suggestion
    logger.info(f'Saved all output to: {outdir}')
```

File: bin/dm_mripicture.py [hunk 9]
```diff
@@ -40,20 +50,35 @@ def main():
...
+    if debug:
+        logger.setLevel(logging.DEBUG)
+    if quiet:
+        logger.setLevel(logging.ERROR)
+
     if subs:
-        print('Creating brain pictures for subjects [', ', '.join(subs),
-              '] from', study, 'project using', tag, 'scans.')
+        logger.info('Creating pictures for subjects [ {} ] from {} project '
+                    'using {} scans.'.format(', '.join(subs), study, tag))
```
Comment (josephmje) [inline bin/dm_mripicture.py:68]:
```suggestion
        logger.info(f'Creating pictures for subjects [ {', '.join(subs)} ] from {study} project '
                    f'using {tag} scans.')
```

**Review by josephmje — COMMENTED** [review]
Thanks for the update @kyjimmy! I just have some minor suggestions about using f strings instead of format for readability and using swapping `--replace` with `--force`.

**Review by josephmje — DISMISSED** [review]

File: bin/dm_mripicture.py [hunk 10]
```diff
@@ -14,11 +14,17 @@
     -s --subject        Subjects
     -o --output=FOLDER  Output directory
```
Comment (jerdra) [inline bin/dm_mripicture.py:15]: might be a good idea to include the default directory if --output not specified

**Review by jerdra — COMMENTED** [review]
looks good! just a minor nitpick on `--output`

File: bin/dm_mripicture.py [hunk 11]
```diff
@@ -27,33 +33,69 @@
...
+    force = arguments["--force"]
+    quiet = arguments["--quiet"]
+    debug = arguments["--debug"]
+    verbose = arguments["--verbose"]
     config = datman.config.config(study=study)
 
+    # setup logging
+    ch = logging.StreamHandler(sys.stdout)
+    ch.setLevel(logging.WARN)
+    logger.setLevel(logging.WARN)
```
Comment (jerdra) [inline bin/dm_mripicture.py:61]: is there a reason to prefer this over basicConfig?

File: bin/dm_mripicture.py (hunk 10 above)
Comment (kyjimmy) [reply bin/dm_mripicture.py:15]: `-o --output=FOLDER  Output directory (default directory: /archive/data/{study}/data/tshirt)`
is this okay?

Comment (jerdra) [reply bin/dm_mripicture.py:15]: yep!

File: bin/dm_mripicture.py (hunk 11 above)
Comment (josephmje) [reply bin/dm_mripicture.py:61]: my bad, i was using the example of `dm_xnat_extract.py` but i think we've moved passed this. @DESm1th suggests only including the StreamHandler for scripts that we submit to the queue.

File: bin/dm_mripicture.py [hunk 12]
```diff
@@ -27,33 +33,69 @@
 import datman.scan
 
 
+logger = logging.getLogger(os.path.basename(__file__))
+
```
Comment (josephmje) [inline bin/dm_mripicture.py:37]:
```suggestion
logging.basicConfig(level=logging.WARN,
                    format="[%(name)s] %(levelname)s: %(message)s")
logger = logging.getLogger(os.path.basename(__file__))

```

File: bin/dm_mripicture.py (hunk 11 above)
Comment (josephmje) [inline bin/dm_mripicture.py:61]:
```suggestion
```

File: bin/dm_mripicture.py [hunk 13]
```diff
@@ -27,33 +33,69 @@
...
     config = datman.config.config(study=study)
 
+    # setup logging
+    ch = logging.StreamHandler(sys.stdout)
+    ch.setLevel(logging.WARN)
+    logger.setLevel(logging.WARN)
+
+    if quiet:
+        logger.setLevel(logging.ERROR)
+        ch.setLevel(logging.ERROR)
```
Comment (josephmje) [inline bin/dm_mripicture.py:65]:
```suggestion
```

File: bin/dm_mripicture.py [hunk 14]
```diff
@@ -27,33 +33,69 @@
...
+    if verbose:
+        logger.setLevel(logging.INFO)
+        ch.setLevel(logging.INFO)
```
Comment (josephmje) [inline bin/dm_mripicture.py:68]:
```suggestion
```

File: bin/dm_mripicture.py [hunk 15]
```diff
@@ -27,33 +33,69 @@
...
+    if debug:
+        logger.setLevel(logging.DEBUG)
+        ch.setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter(
+        f"%(asctime)s - %(name)s - {study} - %(levelname)s - %(message)s"
+    )
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
```
Comment (josephmje) [inline bin/dm_mripicture.py:77]:
```suggestion
```

File: bin/dm_mripicture.py (hunk 11 above)
Comment (DESm1th) [reply bin/dm_mripicture.py:61]: Whoops, I think I miscommunicated XD What I meant was, basicConfig is best if you're just adding a stream handler which is 99% of our logging cases. The only time it's ever really worth manually adding a handler is if you're using filters or want to add a different type of handler, like a SocketHandler to log to our log server. So I agree with Jer, but the reasons got a bit confused haha.

**Review by jerdra — APPROVED** [review]
Good work!

**Review by josephmje — APPROVED** [review]

**Pull request merged**